StringCalculator/StringCalculator.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+
+namespace StringCalculator
+{
+    public class StringCalculator
+    {
+        public StringCalculator()
+        {
+
+        }
+
+        public int Add(string stringInput)
+        {
+            var total = 0;
+            if (String.IsNullOrEmpty(stringInput)) return total;
+
+            if (stringInput.Contains("//"))
+            {
+                var delimiterList = CreateDelimiterList(stringInput);
+                var stringToSum = ExtractStringToSum(stringInput);
+                string[] delimiterArray = delimiterList.ToArray();
+                var stringNumbers = stringToSum.Split(delimiterArray, StringSplitOptions.None);
+                total = SumStringNumbers(stringToSum, total, delimiterArray);
+            }
+            else
+            {
+                var delimiterList = CreateDelimiterList();
+                string[] delimiterArray = delimiterList.ToArray();
+                total = SumStringNumbers(stringInput, total, delimiterArray);
+            }
+
+           return total;
+        }
+
+        public int SumStringNumbers(string text, int total, string[] delimiterArray)
```

Comment:
All these internal methods can be made private. Ideally, you want all your public methods tested.